### PR TITLE
Add recreate flag for start cmd

### PR DESCRIPTION
**Pull request by Junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

Fixes: https://github.com/lima-vm/lima/issues/743

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,6 +194,11 @@ To create an instance "default" from a remote URL (use carefully, with a trustab
 $ limactl start --name=default https://raw.githubusercontent.com/lima-vm/lima/master/examples/alpine.yaml
 ```
 
+To recreate an existing instance:
```
Comment (jandubois) [inline README.md:197]: Maybe it is just me, but I would want to be explicit that this action will delete the existing instance. In some ways it is implied, but I think it doesn't hurt to spell it out that all content is lost:

```suggestion
To destroy an instance and recreate it with identical settings (resets the content of the instance):
```

**Review by jandubois — COMMENTED** [review]
The `--recreate` option should either stop a running instance, or issue an error message that tells the user to stop the instance before it can be recreated. The current code simply ignores the `--recreate` option:

```console
$ limactl start default --recreate
INFO[0000] Using the existing instance "default"
INFO[0000] The instance "default" is already running. Run `lima` to open the shell.
```

**Comment by AkihiroSuda** [comment]
On second thought, maybe we should have this as `limactl factory-reset INSTANCE` rather than overloading `limactl start` command?

**Comment by jandubois** [comment]
> maybe we should have this as `limactl factory-reset INSTANCE` rather than overloading `limactl start` command?

I'm fine with this PR now, but I also prefer to use a separate command. We already overload `start` with 2 meanings:

1. Create a new instance and start it
2. Start an existing instance

So maybe adding a 3rd meaning which will actually destroy (and then recreate) an instance is a step too far. 😄 

Also the word "factory-reset" makes the destructive nature much clearer than the `--recreate` parameter, at least in my mind.

**Review by jandubois — APPROVED** [review]
Approving, because the PR is fine now, but I won't merge because I also prefer to use `limactl factory-reset` instead.

**Comment by Junnplus** [comment]
How about adding a `limactl reset` command? @AkihiroSuda @jandubois

**Comment by AkihiroSuda** [comment]
> How about adding a `limactl reset` command? @AkihiroSuda @jandubois

That may surprise some users who except `reset` to just reboot the instance without wiping the data

**Comment by AkihiroSuda** [comment]
Maybe `limactl reset-data`?

**Comment by jandubois** [comment]
> Maybe `limactl reset-data`?

I still prefer "factory reset". It may feel a bit weird applied to pure software, but I think everybody intuitively understands it means "reset everything as if you just opened the box", which is precisely what it does.

**Comment by Junnplus** [comment]
I remade a PR https://github.com/lima-vm/lima/pull/757 for `limactl factory-reset`, close the current.

**Pull request closed without merging**